exec/java-exec/src/main/resources/rest/storage/list.ftl

```diff
@@ -17,79 +17,280 @@
+              <label>
+                <input type="radio" name="group" id="enabled" value="enabled">
+                ENABLED
+              </label>
+            </div>
+            <div class="radio">
+              <label>
+                <input type="radio" name="group" id="disabled" value="disabled">
+                DISABLED
+              </label>
+            </div>
+          </div>
+        </div>
+
+        <div class="modal-footer">
+          <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
+          <button type="button" id="export" class="btn btn-primary">Export</button>
+        </div>
+      </div>
+    </div>
   </div>
-  <div>
-    <h4>New Storage Plugin</h4>
-    <form class="form-inline" id="newStorage" role="form" action="/" method="GET">
-      <div class="form-group">
-        <input type="text" class="form-control" id="storageName" placeholder="Storage Name">
+  <#-- Modal window for exporting plugin config (including group plugins modal) -->
+
+  <#-- Modal window for creating plugin -->
+  <div class="modal fade" id="new-plugin-modal" role="dialog" aria-labelledby="configuration">
+    <div class="modal-dialog" role="document">
+      <div class="modal-content">
+        <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+          <h4 class="modal-title" id="configuration">New Storage Plugin</h4>
+        </div>
+        <div class="modal-body">
+
+          <form id="createForm" role="form" action="/storage/create" method="POST">
+            <input type="text" class="form-control" name="name" placeholder="Storage Name">
+            <h3>Configuration</h3>
+            <div class="form-group">
+              <div id="editor" class="form-control"></div>
+                <textarea class="form-control" id="config" name="config" data-editor="json" style="display: none;">
+                </textarea>
+            </div>
+            <div style="text-align: right; margin: 10px">
+              <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
+              <button type="submit" class="btn btn-primary" onclick="doCreate()">Create</button>
+            </div>
+          </form>
+
+          <div id="message" class="hidden alert alert-info">
+          </div>
+        </div>
       </div>
-      <button type="submit" class="btn btn-default" onclick="doSubmit()">Create</button>
-    </form>
+    </div>
   </div>
+  <#-- Modal window for creating plugin -->
+
   <script>
-    function doSubmit() {
-      var name = document.getElementById("storageName");
-      var form = document.getElementById("newStorage");
-      form.action = "/storage/" + name.value;
-      form.submit();
-    };
     function doEnable(name, flag) {
-      $.get("/storage/" + name + "/enable/" + flag, function(data) {
+      $.get("/storage/" + name + "/enable/" + flag, function() {
```

Comment:
There should be a check if the flag is `false` ; i.e. an attempt is being made to disable a plugin... which should be followed by a confirmation first, before an attempt to disable does occur.

Reply:
Done in similar way, how it is done in other pages

Reply:
I see the alert box popping up. We should be standardized to using modals whenever we are making UI enhancements.

Reply:
Agreed. Let's not block this PR for that.

Reply:
I have created ticket for this task:
https://issues.apache.org/jira/browse/DRILL-7134